pkg/console/subresource/route/route.go

```diff
@@ -35,6 +36,137 @@ type CustomTLSCert struct {
 	Key         string
 }
 
+type RouteConfig struct {
+	defaultRoute RouteControllerSpec
+	customRoute  RouteControllerSpec
+	domain       string
+	routeName    string
+}
+
+type RouteControllerSpec struct {
+	hostname   string
+	secretName string
+}
+
+func getComponentRouteSpec(ingressConfig *configv1.Ingress, componentName string) *configv1.ComponentRouteSpec {
+	for i, componentRoute := range ingressConfig.Spec.ComponentRoutes {
+		if componentRoute.Name == componentName && componentRoute.Namespace == api.OpenShiftConsoleNamespace {
+			componentRoute := ingressConfig.Spec.ComponentRoutes[i].DeepCopy()
+			return componentRoute
+		}
+	}
+	return nil
+}
+
+func getComponentRouteStatus(ingressConfig *configv1.Ingress, componentName string) *configv1.ComponentRouteStatus {
+	for i, componentRoute := range ingressConfig.Status.ComponentRoutes {
+		if componentRoute.Name == componentName && componentRoute.Namespace == api.OpenShiftConsoleNamespace {
+			componentRoute := ingressConfig.Status.ComponentRoutes[i].DeepCopy()
+			return componentRoute
+		}
+	}
+	return nil
+}
+
+func NewRouteConfig(operatorConfig *operatorv1.Console, ingressConfig *configv1.Ingress, routeName string) *RouteConfig {
+	defaultRoute := RouteControllerSpec{
+		hostname: GetDefaultRouteHost(routeName, ingressConfig),
+	}
+	var customRoute RouteControllerSpec
+	var isIngressConfigCustomHostnameSet bool
+
+	// Custom hostname in ingress config takes precedent over console operator's config
+	componentRouteSpec := getComponentRouteSpec(ingressConfig, routeName)
+	if componentRouteSpec != nil {
+		customRoute.hostname = string(componentRouteSpec.Hostname)
```

Comment:
You may want to validate the hostname.

Reply:
All the validation hostname + secret is done in the route controller. This is just setting an internal config.